src/app/core/services/rundown-timing.service.ts

```diff
@@ -0,0 +1,125 @@
+import { BackwardRundownTiming, RundownTiming } from '../models/rundown-timing'
+import { RundownTimingType } from '../enums/rundown-timing-type'
+import { Rundown } from '../models/rundown'
+import { Segment } from '../models/segment'
+import { PartEntityService } from './models/part-entity.service'
+import { Part } from '../models/part'
+
+export class RundownTimingService {
+  private readonly partEntityService: PartEntityService = new PartEntityService()
+
+  public getExpectedStartEpochTime(rundownTiming: RundownTiming): number | undefined {
+    switch (rundownTiming.type) {
+      case RundownTimingType.FORWARD:
+        return rundownTiming.expectedStartEpochTime
+      case RundownTimingType.BACKWARD:
+        return this.getExpectedStartFromBackwardRundownTiming(rundownTiming)
+      default:
+        return undefined
+    }
+  }
+
+  private getExpectedStartFromBackwardRundownTiming(backwardRundownTiming: BackwardRundownTiming): number | undefined {
+    if (backwardRundownTiming.expectedStartEpochTime) {
+      return backwardRundownTiming.expectedStartEpochTime
+    }
+
+    if (backwardRundownTiming.expectedDurationInMs) {
+      return backwardRundownTiming.expectedEndEpochTime - backwardRundownTiming.expectedDurationInMs
+    }
+
+    return undefined
+  }
+
+  public getExpectedDurationInMs(rundownTiming: RundownTiming): number | undefined {
+    return rundownTiming.expectedDurationInMs
+  }
```

Comment:
Is this method really needed?
Why can't the caller just do `rundownTiming.expectedDurationInMs` themselves?

Reply:
I see how it might look weird, but it is to not mix up the way we handle `RundownTiming` when for instance it is used by `BasicRundowns`. 
When they need the planned end or start (located in the `RundownTiming`), we first have to check which type of time calculation we are dealing with: `FORWARD`, `BACKWARD` og `UNSCHEDULED`. Though this is not necessary for the `expectedDurationInMs`, i think it would be less than ideal if we handled some of this inside a service and some of it outside.
We might even change how this is implemented, and in that case it would be nice to have it wrapped in the service.